src/lib/core/CHIPTLVReader.cpp

```diff
@@ -731,10 +731,14 @@ CHIP_ERROR TLVReader::ReadElement()
         break;
     case kTLVFieldSize_8Byte:
         mElemLenOrVal = LittleEndian::Read64(p);
+        VerifyOrExit(!TLVTypeHasLength(elemType) || (mElemLenOrVal <= UINT32_MAX), err = CHIP_ERROR_NOT_IMPLEMENTED);
```

Comment:
```suggestion
        VerifyOrReturnError(!TLVTypeHasLength(elemType) || (mElemLenOrVal <= UINT32_MAX), CHIP_ERROR_NOT_IMPLEMENTED);
```
and then you don't need to mess with the `exit:` bits...